cli: TestZipIncludeAndExcludeFilesDataDriven failed

cli.TestZipIncludeAndExcludeFilesDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18410178?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18410178?buildTab=artifacts#/) on master @ [ec6e761a7dc4df7f5f3fc90e53cde8f3c48d4e21](https://github.com/cockroachdb/cockroach/commits/ec6e761a7dc4df7f5f3fc90e53cde8f3c48d4e21):


```
debug/system.jobs.txt
debug/system.lease.txt
debug/system.locations.txt
debug/system.migrations.txt
debug/system.namespace.txt
debug/system.privileges.txt
debug/system.protected_ts_meta.txt
debug/system.protected_ts_records.txt
debug/system.rangelog.txt
debug/system.replication_constraint_stats.txt
debug/system.replication_critical_localities.txt
debug/system.replication_stats.txt
debug/system.reports_meta.txt
debug/system.role_id_seq.txt
debug/system.role_members.txt
debug/system.role_options.txt
debug/system.scheduled_jobs.txt
debug/system.settings.txt
debug/system.span_configurations.txt
debug/system.sql_instances.txt
debug/system.sql_stats_cardinality.txt
debug/system.sqlliveness.txt
debug/system.statement_diagnostics.txt
debug/system.statement_diagnostics_requests.txt
debug/system.statement_statistics_limit_5000.txt
debug/system.table_statistics.txt
debug/system.task_payloads.txt
debug/system.tenant_settings.txt
debug/system.tenant_tasks.txt
debug/system.tenant_usage.txt
debug/system.tenants.txt
debug/tenant_ranges.err.txt
    zip_test.go:1453: -- test log scope end --
    datadriven.go:144: 
        testdata/zip/file-filters/testzip_file_filters:2:
         
        output didn't match expected:
        @@ -66,10 +66,11 @@
         debug/nodes/1/crdb_internal.node_txn_execution_insights.txt
         debug/nodes/1/crdb_internal.node_txn_stats.txt
         debug/nodes/1/details.json
         debug/nodes/1/gossip.json
         debug/nodes/1/heap.pprof
        +debug/nodes/1/heapprof/memprof.2025-01-07T18_16_21.428.132126856.pprof
         deterministic.log
         debug/nodes/1/lsm.txt
         debug/nodes/1/ranges.json
         debug/nodes/1/stacks.txt
         debug/nodes/1/stacks_with_labels.txt
    --- FAIL: TestZipIncludeAndExcludeFilesDataDriven/testzip_file_filters (22.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipIncludeAndExcludeFilesDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46161

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -1376,7 +1376,7 @@ func trimNonDeterministicZipOutputFiles(out string) string {
 	out = re.ReplaceAllString(out, ``)
 	re = regexp.MustCompile(`(?m).*job_status\.txt$` + "\n")
 	out = re.ReplaceAllString(out, ``)
-	re = regexp.MustCompile(`(?m).*(memprof|memstats|memmonitoring).*\.txt$` + "\n")
+	re = regexp.MustCompile(`(?m).*(memprof|memstats|memmonitoring).*\.(txt|pprof)$` + "\n")
 	out = re.ReplaceAllString(out, ``)
 	re = regexp.MustCompile(`(?m).*goroutine_dump.*\.txt\.gz$` + "\n")
 	out = re.ReplaceAllString(out, ``)

